### ISSUE TITLE
For extending the type alias created with `TypeAlias` or an assignment statement, pyright should also give error for consistency

### ISSUE THREAD

**Issue by hyperkai** [microsoft/pyright#11201]
Memo:
- pyright test.py
- pyright 1.1.407
- Python 3.14.0

Extending the type alias created with a [type statement](https://docs.python.org/3/reference/simple_stmts.html#type) or [TypeAliasType](https://docs.python.org/3/library/typing.html#typing.TypeAliasType), pyright gives the error as shown below:

```python
''' Type statement '''
type TA = int
''' Type statement '''

''' TypeAliasType '''
# from typing import TypeAliasType

# TA = TypeAliasType('TA', value=int)
''' TypeAliasType '''

class Cls(TA): pass # Error
```

> error: A type alias defined in a "type" statement cannot be used as a base class

But extending the type alias created with [TypeAlias](https://docs.python.org/3/library/typing.html#typing.TypeAlias) or an [assignment statement](https://docs.python.org/3/reference/simple_stmts.html#assignment-statements), pyright doesn't give error as shown below so error should also occur for consistency:

```python
''' TypeAlias '''
from typing import TypeAlias

TA: TypeAlias = int
''' TypeAlias '''

''' Assignment statement '''
# TA = int
''' Assignment statement '''

class Cls(TA): pass # No error
```

**Comment by erictraut** [comment]
Pyright is working correctly here.

There's no such thing as "extending a type alias". What you're doing here is using a symbol in a class declaration statement. The use of a `TypeAliasType` instance in a class declaration statement results in a runtime exception, so pyright is correct to report that as an error. The use of a type alias defined using the `TypeAlias` special form does not result in a runtime exception when it's used in a class declaration, so a type checker should not flag it as an error.
